pkg/policy/k8s/service.go

```diff
@@ -39,12 +39,13 @@ func (p *policyWatcher) onServiceEvent(event k8s.ServiceNotification) {
 // added, removed, its endpoints have changed, or its labels have changed).
 // This function then checks if any of the known CNP/CCNPs are affected by this
 // change, and recomputes them by calling resolveCiliumNetworkPolicyRefs.
-func (p *policyWatcher) updateToServicesPolicies(svcID k8s.ServiceID, newSVC, oldSVC *k8s.Service) error {
+func (p *policyWatcher) updateToServicesPolicies(svcID k8s.ServiceID, newSVC, oldSVC *k8s.MinimalService, newEps, oldEps *k8s.MinimalEndpoints) error {
 	var errs []error
+	endpointsChanged := !newEps.DeepEqual(oldEps)
```

Comment:
This might've uncovered an existing bug. We were only checking the service for equality, but we actually also cared about the endpoints changing. I expect this was the cause for this failure: https://github.com/cilium/cilium/actions/runs/13389227688/job/37393038062 (to be checked...). Not sure why exactly this worked before when it was comparing the more complex `*Service` for equality, guess something in there changed when endpoints changed.

Reply:
does it make sense to add a zero-sized uncomparable field in the struct? Hopefully no more demons are awoken :see_no_evil: